### DISCUSSION TITLE
Ability to abort the current transaction and unlock the plug

### DISCUSSION THREAD

**Discussion by emarvegt** [leeyuentuen/alfen_wallbox#74] [Ideas]
I have set a very long timeout and “continue” when disconnecting the plug on EV side, so the transaction resumes when I return home and I don’t have to do the unplug/plug and authenticate routine every time. This works mostly fine, but sometimes the chargepoint gets into a state where it won’t resume but won’t unlock or abort either. Even a reboot doesn’t help. In that case it would be great if I could ‘kill’ the current transaction in HA. I know some OCPP back-ends have this option. Not sure it can be done through the local connection?

**Comment by leeyuentuen** [comment]
i think  there is here 2 question

the first one is the timeout when cable is connected, this is to prevent that you give the authentication for the transaction without plugging in the cable or forgot. so that someone else won't plug that cable and use your authenticated card to continue charge

the second question here is to end the current transaction, what happened if you just set the wallbox to in-operation

**Reply by emarvegt** [reply]
Regarding the timeout, in the ACE tool I can set the timeout length and what it will do within the timeout period: 
Continue) will result the transaction if you unplug and re-plug within the timeout limit, which is what I used to keep a transaction going even when I was going to work and come back at the end of the day.
Abort Unlock) unlocks the plug so when I return I have to unplug, replug and then authenticate to start charging.
Abort Lock) keeps the plug locked but it doesn’t seem to unlock when I then authenticate again, nor does it start charging when I do.
Abort unlock when offline) same as above, only when connection to the backend is not available.

When the chargepoint gets in this stuck state, the plug remains locked but not charging and gives an error when I authenticate (with the same tag I always use). However, the status stays on “in operation” during that time. And that stuck state persists across a software-initiated reboot, but not when I cut power to the chargepoint.

**Reply by leeyuentuen** [reply]
i'll take a look in ace  i think i didn't check that part from the ace installer

**Comment by leeyuentuen** [comment]
@emarvegt should it be one of these settings?

**Reply by emarvegt** [reply]
@leeyuentuen I have not found a way to abort the currently running charge session from the ACE installer either. I only know OCPP supports it because I read it in the documentation somewhere.

**Comment by leeyuentuen** [comment]
@emarvegt  this is alfen ocpp supported page, maybe you can see which one?
[alfen ocpp.pdf](https://github.com/leeyuentuen/alfen_wallbox/files/13126631/alfen.ocpp.pdf)

**Comment by emarvegt** [comment]
@leeyuentuen in that list, I don’t find any parameter that could stop a running transaction. In fact, there is no parameter there that gives any info about running transactions either, which is strange, right? At minimum OCPP transmits from the chargepoint to the back end some transaction details: start time, end time, delivered energy. After all, the back end needs that info to send the user a monthly overview.
So maybe there is an additional set of standard OCPP instructions not included in that list?
